### ISSUE TITLE
mi and mo in TeX to MathML conversion [was: TeX command \dim should be rendered as an operator in MathML]

### ISSUE THREAD

**Issue by physikerwelt** [mathjax/MathJax#1309]
See discussion in #1305

**Comment by pkra** [comment]
Should is a strong word. Personally, I don't think it should (it's just a function name which is what `mi` is for) but I'm not too hung up on it. 

In any case, this could not change without a major version jump since it would break (lots of) existing content. 

Why not redefine `\dim` with `\mmlToken{mo}{dim}` in `mediawiki-texvc.js`?

**Comment by physikerwelt** [comment]
@pkra I would love to do that. I would really aprerchiate if you could help to define that, since I have no clue how to get this into the source code.

**Comment by pkra** [comment]
I think the following should be enough:

```
dim: ["Macro", "\\mmlToken{mo}{dim}"],
```

(and add `<li>dim:$\dim$</li>` to `test/sample-mediawiki-texvc.html` I suppose).

**Comment by physikerwelt** [comment]
@pkra Thank you very much.
But, I realized that the problem affects all kind of functions. e.g. \sin\acrsin x
See
http://codepen.io/physikerwelt/pen/wKZQby
So I'm not sure if that nice additional definition solves the problem.

**Comment by physikerwelt** [comment]
@pkra You are absolutly right. According to the MathML spec mi is more appropriate for dim. (http://www.w3.org/TR/MathML2/chapter4.html)

**Comment by pkra** [comment]
> But, I realized that the problem affects all kind of functions.

Yes.

> So I'm not sure if that nice additional definition solves the problem.

Right. Cf. `namedFn`,
https://github.com/mathjax/MathJax/blob/ba9afeb5a743249acdea17540b48b14ebc95dbe4/unpacked/jax/input/TeX/jax.js#L765-L796

But hold off on changing anything until Davide has time to respond. The choice to differentiate MathML's and TeX's notion of operator is deliberate so changing this is bound to have side effects. I'll make a note to bring it up on our next developer meeting.

**Comment by physikerwelt** [comment]
@pkra Can you adjust the title of the conversation to keep the issue tracker tracable for a wide audience. While I see that we are making progress towards resolving #1305, I think my wrong assumption that mo would be to correct rendering might be confusion for others.

**Comment by physikerwelt** [comment]
ok I'll wait for @dpvc

**Comment by pkra** [comment]
>  Can you adjust the title of the conversation to keep the issue tracker tracable for a wide audience. 

Happy to. Any suggestions?

> I think my wrong assumption that mo would be to correct rendering might be confusion for others.

I don't think you're wrong, it's more a matter of opinion. E.g., LaTeXML creates `mo`s for `\dim` (even for `\mathrm{ker}`). 

How about I tag this as question and make the title "`mi` and `mo` in TeX to MathML conversion"?

**Comment by physikerwelt** [comment]
I would be interested what @brucemiller and @dginev think about that.
For me the examples in http://www.w3.org/TR/MathML3/chapter3.html#presm.mi suggest to use mi rather than mo.

**Comment by pkra** [comment]
Cross link to @dpvc's comment at https://github.com/mathjax/MathJax/issues/1305#issuecomment-158961615

**Comment by davidcarlisle** [comment]
The advice in different version of mathml going back to the start has been a bit variable on this but finally in MatHML3 spec we came to the conclusion that multi-letter identifiers should be marked as mi and all examples of presentation mathml renderings of content mathml now use that form so 

`<apply><sin/>` 

becomes 

`<mrow><mi>sin</mi><mo>&#x2061;</mo>...`

with the invisible apply operator handling the spacing requirements.

**Comment by brucemiller** [comment]
It was originally true that LaTeXML generated mo for ```\sin``` and similar, but that was changed to mi a long time ago; probably when this came up previously in the Math WG and the advice was clarified.  There'll usually  be (pending successful parsing of the math) an InvisibleApply following it, which theoretically should handle spacing.

**Comment by dginev** [comment]
@brucemiller ```\dim``` is still an ```mo``` today however, probably that needs to be updated to an mi as well? @pkra also mentioned that ```\mathrm{ker}``` is an mo, though when I test I see an ```mi```. ([quick reference here](http://latexml.mathweb.org/editor?tex=%24%24%5Cdim%20x%24%24%0A%0A%24%24%5Cmathrm%7Bker%7D%20x%20%24%24))

It's nice to see a consensus that "multi-letter identifiers" are to be consistently represented as ```mi```s, that makes things simple.

**Comment by pkra** [comment]
@dginev thanks for correcting me regarding `ker` -- `mi` it is (I was using the same demo page so I must have mistyped earlier).

**Comment by brucemiller** [comment]
On 11/23/2015 02:28 PM, Deyan Ginev wrote:
> @brucemiller <https://github.com/brucemiller> |\dim| is still an |mo|
> today however, probably that needs to be updated to an mi as well? @pkra
> <https://github.com/pkra> also mentioned that |\mathrm{ker}| is an mo,
> though when I test I see an |mi|. (quick reference here
> <http://latexml.mathweb.org/editor?tex=%24%24%5Cdim%20x%24%24%0A%0A%24%24%5Cmathrm%7Bker%7D%20x%20%24%24>)
>
> It's nice to see a consensus that "multi-letter identifiers" are to be
> consistently represented as |mi|s, that makes things simple.

Hmm... maybe concensus, maybe not :>

I have \dim marked as a "limitop" (along with \lim, \liminf, etc),
which are noted for getting their subscripts underneath, and
are more operator-like in their syntax (apply to expressions w/o 
parentheses, etc).
Not that this by itself forces an mo over an mi, but it raises
the psychological barrier.

**Comment by pkra** [comment]
Closing this in favor of #1309. Feel free to continue the discussion though.

**Comment by physikerwelt** [comment]
@pkra  do you mean #1312?

**Comment by pkra** [comment]
Oh, thanks for catching that! Yes, I meant #1312.

**Comment by brucemiller** [comment]
Makes sense to close from MathJax POV, but if you don't mind our hijacking your thread, it would be nice for us to be all on the same (or similar) page.

The MathML spec explicitly calls out "lim" as an operator for which movablelimits is appropriate, which is not available for mi.  So, "lim", at least,  is a "multi-letter identifier" that should be represented by mi (unless that clause in the spec is a leftover?).  Perhaps it's debatable whether ```\dim``` is in that same class.

[Aside: LaTeXML always disables movablelimits based on the rest of the markup, explicitly choosing subsup or overunder depending. But that's only because we think we know what you wanted]

**Comment by pkra** [comment]
>  if you don't mind our hijacking your thread, it would be nice for us to be all on the same (or similar) page.

I don't think anyone minds. But I don't think this thread will reach everyone. Maybe start something on www-math?
